### PR TITLE
docs: define how the default CMM uses a master key provider

### PR DESCRIPTION
**Pull request by mattsb42-aws**
*Issue #, if available:*
resolves: #98 

*Description of changes:*

This addition defines how the default CMM uses a master key provider.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: framework/default-cmm.md [hunk 1]
```diff
@@ -55,7 +59,10 @@ If the algorithm suite contains a [signing algorithm](algorithm-suites.md#signat
   - The key MUST be the reserved name, `aws-crypto-public-key`.
   - The value MUST be the base64-encoded public verification key.
 
-On each call to Get Encryption Materials, the default CMM MUST make a call to the key provider's [On Encrypt](keyring-interface.md#onencrypt) function.
+On each call to Get Encryption Materials,
+the default CMM MUST make a call to the keyring's [On Encrypt](keyring-interface.md#onencrypt) function.
+If the default CMM is configured with a master key provider instead,
```
Comment (seebees) [inline framework/default-cmm.md:64]: I would make it clear here that master key provider is both legacy
and not implemented in every runtime here as well.

Or, we could leave this out
and have the description of mater key providers
be the only description.
In that case, I would suggest a callout in the Overview.

File: framework/default-cmm.md [hunk 2]
```diff
@@ -69,10 +76,53 @@ The values obtained above MUST be included in the encryption materials returned.
 If the algorithm suite contains a [signing algorithm](algorithm-suites.md#signature-algorithm),
 the default CMM MUST remove the verification key from the encryption context.
 
-On each call to Decrypt Materials, the default CMM MUST make a call to the key provider's [On Decrypt](keyring-interface.md#ondecrypt) function.
+On each call to Decrypt Materials,
+the default CMM MUST make a call to the keyring's [On Decrypt](keyring-interface.md#ondecrypt) function.
+If the default CMM is configured with a master key provider instead,
+see [Decrypt Materials (master key provider)](#decrypt-materials-master-key-provider).
+
+The default CMM MUST obtain the following from the response:
```
Comment (seebees) [inline framework/default-cmm.md:84]: Duplicate?

File: framework/default-cmm.md (hunk 1 above)
Comment (lavaleri) [reply framework/default-cmm.md:64]: +1
I think I'd prefer the latter, but at the very least if we mention MPKs in the main spec let's make it more clear that this is referring to legacy behavior that one SHOULD NOT implement in new implementations.

File: framework/default-cmm.md (hunk 2 above)
Comment (mattsb42-aws) [reply framework/default-cmm.md:84]: yup

File: framework/default-cmm.md (hunk 1 above)
Comment (mattsb42-aws) [reply framework/default-cmm.md:64]: Yeah, I think I'm ok with that. If someone knows they're looking for MKP behavior, they shouldn't have any issues finding it in the "Legacy" section, and the descriptions in that section make it clear where they fit into the spec.

**Pull request merged**